getyour/app/views/application.py

```diff
@@ -135,36 +148,38 @@ def account(request):
             # AJAX data function below, sends data to AJAX function in account.html, if clients make a mistake via email or phone number, page lets them know and DOESN'T refresh web page
             # let's them know via AJAX
             error_messages = dict(form.errors.items())
-            
+
             # Create message_dict by parsing strings or all items of ErrorList,
             # where applicable. Use the prettified field name as each key
             message_dict = {}
             for keyitm in error_messages.keys():
                 val = error_messages[keyitm]
 
                 # Gather the list of error messages and flatten it
-                message_list = [[y for y in val] if isinstance(val, ErrorList) else val]
-                flattened_messages = [item for items in message_list for item in items]
+                message_list = [[y for y in val]
+                                if isinstance(val, ErrorList) else val]
+                flattened_messages = [
+                    item for items in message_list for item in items]
 
                 # Write the messages as a string
                 message_dict.update({
                     keyitm.replace('_', ' ').title(): '. '.join(flattened_messages)
-                    })
+                })
             # Create error message data by prepending the prettified field name
             # and joining as newlines
             data = {
                 'result': 'error',
                 'message': '\n'.join(
                     [f"{keyitm}: {message_dict[keyitm]}" for keyitm in message_dict.keys()]
-                    )
-                }
+                )
+            }
             return JsonResponse(data)
     else:
-        if update_mode:
-            # Query the users table for the user's data
+        try:
             user = User.objects.get(id=request.user.id)
             form = UserUpdateForm(instance=user)
-        else:
+            update_mode = True
```

Comment:
Yeah this is a little confusing. The reason for this `update_mode` getting set here (instead of up top) is in the instance of a user navigating back to the `account` step. This way the user can update their account information.

Reply:
Ah, I see. And it's in this function only because changing data on a non-update account page would create a new account, which wouldn't be the intended action. Perfect, thanks!